server/src/main/java/org/opensearch/search/internal/ContextIndexSearcher.java

```diff
@@ -257,7 +258,15 @@ public void search(
 
     @Override
     public void search(Query query, Collector collector) throws IOException {
-        super.search(query, collector);
+        // TODO : Remove when switching to use the @org.apache.lucene.search.IndexSearcher#search(Query, CollectorManager) variant from
+        // @org.opensearch.search.query.QueryPhase#searchWithCollector which then calls the overridden
+        // search(LeafReaderContextPartition[] partitions, Weight weight, Collector collector)
+        query = collector.scoreMode().needsScores() ? rewrite(query) : rewrite(new ConstantScoreQuery(query));
```

Comment:
@msfroh this change make sense to me to restore `searchContext.shouldUseTimeSeriesDescSortOptimization()`  optimization, wdyt?